lib/lists.nix

```diff
@@ -35,13 +40,25 @@ rec {
         else op (elemAt list n) (fold' (n + 1));
     in fold' 0;
 
-  /* Left fold: `fold op nul [x_1 x_2 ... x_n] == op (... (op (op nul
-     x_1) x_2) ... x_n)'.
+  /* fold is an alias of foldr for historic reasons */
+  # FIXME(Profpatsch): deprecate?
+  fold = foldr;
```

Comment:
Should probably be clarified in the docs, yes. It’s a non-trivial question that depends on the evaluation characteristics of the runtime.

If the operator is *not* associative, `foldl` and `foldr` will have different results.
Normally it is, though, so it becomes a question of how many thunks each function uses. There’s [an article on HaskellWiki](https://wiki.haskell.org/Foldr_Foldl_Foldl'), but I’m not sure if the same rules hold for our nixpkgs folds.